### DISCUSSION TITLE
UseSWR not returning data from api routes

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/swr#676] [Q&A] [answered]
Hi all,
I don't know what I've done wrong here but useSWR is not pulling the data I would expect from my api routes. Hoping somebody could offer some advice?

I have a page called backorders:
```
const BackOrders = () => {

  const fetcher = url => fetch(url).then(res => res.json())

  const { data, error } = useSwr(`/api/orders`, fetcher)

  console.log(data)

  return <div>Test</div>
}
```

When you navigate to `/api/orders` the expected data is pulled and displayed.

My orders api file:
```
export default async (req, res) => {
  const { complete } = req.query

  if (!complete) {
    const response = await getAllOrders()

    res.json(response.data)
  }

  const response = await getFilteredOrders(complete)

  res.json(response.data)
}
```

I can see the issue in the Axios error: '/Order.json?complete=undefined', complete=undefined should not occur due to my if statement in the orders api file?

Any ideas?

**Comment by sergiodxa** [comment, marked as answer]
Because you are not adding a return at the end of your `if` condition, the rest of the code it's being executed. You need to either return inside the `if` or wrap the rest of the code in an `else`.

**Reply by darrylmorley** [reply]
Doh! Thank you @sergiodxa
